packages/ui/slide-over.tsx

```diff
@@ -0,0 +1,162 @@
+'use client';
+
+import {
+  Fragment,
+  useState,
+  ReactNode,
+  createContext,
+  useContext,
+  Dispatch,
+} from 'react';
+import {Dialog, Transition} from '@headlessui/react';
+import {XMarkIcon} from '@heroicons/react/24/outline';
+import classNames from 'classnames';
+
+interface SlideOverProps {
+  children: ReactNode;
+  variant?: 'text' | 'gallery';
```

Comment:
To be sure: is `variant` allowed to be `undefined`?

Reply:
Yes, it defaults to 'text' because most Slide Overs are the text variant (narrower and white background).